apps/dashboard/src/main/java/com/akto/action/testing/StartTestAction.java

```diff
@@ -343,20 +331,26 @@ public String retrieveAllCollectionTests() {
         if(fetchCicd){
             // filters for test runs to be only CI/CD pipeline
             testingRunFilters.add(Filters.in(Constants.ID, getCicdTests()));
-        } else if(fetchAllTestRuns){
-            // get All test runs which are not run by test editor
-            testingRunFilters.add(Filters.ne("triggeredBy", "test_editor"));
-        } else {
+        } else{
+            testingRunFilters.add(Filters.nin(Constants.ID, getCicdTests()));
+        } 
+        
+        if(fetchOneTime){
+            // get All test runs which are not run by test editor{one time}
+            testingRunFilters.add(Filters.eq("periodInSeconds",0));
+        } else if(fetchRecurring) {
             // the left test are the scheduled one
             Collections.addAll(testingRunFilters, 
-                Filters.lte(TestingRun.SCHEDULE_TIMESTAMP, this.endTimestamp),
-                Filters.gte(TestingRun.SCHEDULE_TIMESTAMP, this.startTimestamp),
-                Filters.nin(Constants.ID,getCicdTests()),
-                Filters.ne("triggeredBy", "test_editor")
+                Filters.ne("periodInSeconds", 0)
             );
         }
 
         testingRunFilters.addAll(prepareFilters());
+        testingRunFilters.add(Filters.and(
+            Filters.lte(TestingRun.END_TIMESTAMP, endTimestamp),
```

Comment:
use scheduled timestamp for this query as endtimestamp might be -1 for tests which haven't started.